src/core/Table/Table.tsx

```diff
@@ -145,6 +145,8 @@ export type TableProps<
    * @default 'default'
    */
   density?: 'default' | 'condensed' | 'extra-condensed';
+  /** Disables row selection when clicked anywhere on the row. */
+  disableOnRowClickSelection?: boolean;
```

Comment:
Maybe call it `selectRowOnClick` (default to `true`).

Reply:
Sounds better. I will change to that.

Reply:
Changed.

Reply:
Update PR title too

Reply:
Done.